lib/components/base-components/Renderable.ts

```diff
@@ -297,9 +299,26 @@ export abstract class Renderable implements IRenderable {
     }
 
     // Check declared async dependencies for this phase within subtree
+    let asyncCheckContext: Renderable = this
+    if ("getSubcircuit" in this) {
+      try {
+        asyncCheckContext = (this as any).getSubcircuit()
+      } catch (e: any) {
+        // If getSubcircuit fails, it's likely because the component is not yet
+        // in a tree with a subcircuit at its root. This is a valid state
+        // (especially in tests), so we fall back to using the component
+        // itself as the context for checking async dependencies.
+        if (!e.message.includes("is not inside an opaque group")) {
+          throw e // Re-throw unexpected errors
+        }
+      }
+    }
     const deps = asyncPhaseDependencies[phase] || []
     for (const depPhase of deps) {
-      if (this._hasIncompleteAsyncEffectsInSubtreeForPhase(depPhase)) return
+      if (
+        asyncCheckContext._hasIncompleteAsyncEffectsInSubtreeForPhase(depPhase)
+      )
+        return
```

Comment:
not sure if this is still in active development, but i think we should revert the changes to this import-to-keep-simple runRenderPhase function